custom/icds_reports/tasks.py

```diff
@@ -1199,6 +1199,47 @@ def build_incentive_report(agg_date=None):
         agg_date = current_month - relativedelta(months=1)
     for state in state_ids:
         AWWIncentiveReport.aggregate(state, agg_date)
+
+    performance_queryset = list(AWWIncentiveReport.objects.values_list('awc_id', 'is_launched',
+                                                                       'valid_visits', 'visit_denominator',
+                                                                       'wer_weighed', 'wer_eligible',
+                                                                       'incentive_eligible')
+                                .order_by('awc_id'))
+    aggregate_queryset = list(AggAwc.objects.values_list('awc_id', 'is_launched',
+                                                         'valid_visits', 'expected_visits')
+                              .order_by('awc_id'))
```

Comment:
Casting the querysets to lists:

1. Will cause everything to be loaded from the DB which could be expensive
2. Makes the variable names ending in `_queryset` very misleading.

Why is it necessary and is there any way to push more logic into the DB instead?

Reply:
we are using the same records in different checks. So i thought loading them into memory and using them in various checks reduces the db calls. Also we are iterating only once for all the checks and passing the resultset rows as parameters

Reply:
Ok, well then at a minimum I think change the variable names.

Do you know how many rows are expected to be returned by this? Only reason it could be an issue is if the results are very large.

Reply:
agreed. we can get them by pagination. Trade off is addition of latency